### PR TITLE
Migrate create transaction commands - Partially closes #558

### LINKED ISSUE #558: Migrate command transaction
### Description

Migrate transaction command

- get
- create
- verify
- sign
- broadcast

### 
Child of #524

### PR DESCRIPTION
**Pull request by shuse2**
### Description
Migration from
```
create transaction transfer
create transaction register second passphrase
create transaction cast votes
create transaction create multisignature account
create transaction register delegate
```

`npm t`
`./bin/run transaction:create`
`./bin/run transaction:create:transfer`
`./bin/run transaction:create:secondpassphrase`
`./bin/run transaction:create:vote`
`./bin/run transaction:create:delegate`
`./bin/run transaction:create:multisignature`

### Review checklist

* It should be merged after #577 
* The PR solves #558 
* All new code is covered with unit tests
* All new code was formatted with Prettier
* Linting passes
* Tests pass
* Commit messages follow the [commit guidelines](CONTRIBUTING.md#git-commit-messages)
* Documentation has been added/updated

### CODE REVIEW

File: src/commands/transaction/create.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+		const clazz =
+			typeClassMap[type in typeNumberMap ? typeNumberMap[type] : type];
+		const resolvedFlags = Object.entries(flags).reduce(resolveFlags, []);
+		await clazz.run([...argv, ...resolvedFlags]);
+	}
+}
+
+CreateCommand.flags = {
+	...BaseCommand.flags,
+	type: flagParser.string({
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/commands/transaction/create.js]: We don't exactly need them, but it's more like an alias
it's always possible to use it like
`transaction:create:transfer` but i think it's better to use it like `transaction:create -t 0`

File: src/commands/transaction/create.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+	passphrase: flagParser.string(commonFlags.passphrase),
+	'second-passphrase': flagParser.string(commonFlags.secondPassphrase),
+	'no-signature': flagParser.boolean(commonFlags.noSignature),
+	votes: flagParser.string(commonFlags.votes),
+	unvotes: flagParser.string(commonFlags.unvotes),
+};
+
+CreateCommand.args = Array(MAX_ARG_NUM)
+	.fill()
+	.map(i => ({
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/commands/transaction/create.js]: When calling the child commands, it seems like we needs to reserve the arg spaces first

File: src/commands/transaction/create.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+	(accumulated, [key, value]) => {
+		accumulated.push(key, value);
+		return accumulated;
+	},
+	[],
+);
+
+const typeClassMap = {
+	transfer: TransferCommand,
+	secondpassphrase: SecondpassphraseCommand,
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/commands/transaction/create.js]: maybe `transaction:create:secondsignature` or `transaction:create:signature` would be better?

File: src/commands/transaction/create/delegate.js [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+ * Removal or modification of this copyright notice is prohibited.
+ *
+ */
+import { flags as flagParser } from '@oclif/command';
+import elements from 'lisk-elements';
+import BaseCommand from '../../../base';
+import commonFlags from '../../../utils/flags';
+import getInputsFromSources from '../../../utils/input';
+
+const processInputs = username => ({ passphrase, secondPassphrase }) =>
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/commands/transaction/create/delegate.js]: I think we are still using HOF in most cases
https://github.com/LiskHQ/lisk-commander/blob/2.0.0/src/commands/passphrase/decrypt.js#L34

File: src/commands/transaction/create.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (willclarktech) [reply src/commands/transaction/create.js]: We've worked hard to remove that whole confusion. The vocabulary makes sense, I'm talking about hyphenated vs single word here.

File: src/commands/transaction/create/transfer.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+		const result = processFunction(inputs);
+		return this.print(result);
+	}
+}
+
+TransferCommand.args = [
+	{
+		name: 'amount',
+		required: true,
+		description: 'Amount of LSK to send.',
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/commands/transaction/create/transfer.js]: Yes, we changed it before 1.0.0

File: src/commands/transaction/create.js [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+const resolveFlags = (accumulated, [key, value]) => {
+	if (key === 'type') {
+		return accumulated;
+	}
+	if (typeof value === 'string') {
+		accumulated.push(`--${key}`, value);
+		return accumulated;
+	}
+	const boolKey = value === false ? `--no-${key}` : `--${key}`;
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/commands/transaction/create.js]: `key` comes in as the flag key.
For `--no-json` case the `key` is `json`, and for `--no-signature` case it's `no-signature`.
so it's passing down the flag to child as it was called with

File: src/commands/transaction/create.js [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+	accumulated.push(boolKey);
+	return accumulated;
+};
+
+export default class CreateCommand extends BaseCommand {
+	async run() {
+		const { argv, flags } = this.parse(CreateCommand);
+		const { type } = flags;
+		const clazz =
```
[the remark this answers is not in the document]
Comment (shuse2) [reply src/commands/transaction/create.js]: renamed to `commandType` after fixing to `Object.Keys()`

File: src/commands/transaction/create.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (willclarktech) [reply src/commands/transaction/create.js]: So we can end up with `--no-no-signature`? We might need to rethink this.

File: src/commands/transaction/create.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (willclarktech) [reply src/commands/transaction/create.js]: Why not make it an argument though?

File: src/commands/transaction/create.js [hunk 8]
```diff
@@ -0,0 +1,97 @@
...
+	votes: flagParser.string(commonFlags.votes),
+	unvotes: flagParser.string(commonFlags.unvotes),
+};
+
+CreateCommand.args = new Array(MAX_ARG_NUM).fill().map(i => ({
+	name: `${i}_arg`,
+}));
+
+CreateCommand.description = `
+Create transaction object.
```
Comment (willclarktech) [inline src/commands/transaction/create.js]: "Creates a"

File: src/commands/transaction/create.js [hunk 9]
```diff
@@ -0,0 +1,97 @@
...
+`;
+
+CreateCommand.examples = [
+	'transaction:create --type=0 100 13356260975429434553L',
+	'transaction:create --type=delegate username',
```
Comment (willclarktech) [inline src/commands/transaction/create.js]: Delegate example also applies here.

File: src/commands/transaction/create.js (hunk 6 above)
[the remark this answers is not in the document]
Comment (shuse2) [reply src/commands/transaction/create.js]: It's not allowed to use `--no-no` and for bool flag, you cannot set it to `false`. it's either --no-signature (true) or false by default

File: src/commands/transaction/create.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (shuse2) [reply src/commands/transaction/create.js]: in this `transaction:create` command, args are reserved for child commands.
that's why this `--type` is flag

**Pull request merged**